plugins/generator-1.18.2/forge-1.18.2/templates/armor.java.ftl

```diff
@@ -63,13 +63,20 @@ public abstract class ${name}Item extends ArmorItem {
 
 			@Override public Ingredient getRepairIngredient() {
 				<#if data.repairItems?has_content>
-				return Ingredient.of(
-							<#list data.repairItems as repairItem>
-								${mappedMCItemToItemStackCode(repairItem,1)}<#if repairItem?has_next>,</#if>
-							</#list>
-				);
+					return Ingredient.fromValues(Stream.of(
+						<#list data.repairItems as item>
+							<#if item.getUnmappedValue().startsWith("TAG:")>
+								new Ingredient.TagValue(ItemTags.create(new ResourceLocation("${item.getUnmappedValue().replace("TAG:", "")}")))
+							<#elseif generator.map(item.getUnmappedValue(), "blocksitems", 1).startsWith("#")>

```

Comment:
This was probably implemented but not tested, as this will keep # character, which will of course then not work.

Please check for other cases of this, fix them and retest

Reply:
This can help on how this is implemented:

![image](https://user-images.githubusercontent.com/16374228/216827281-01f3db76-fff9-40ce-b2e4-c71b4fef487d.png)

Reply:
> This was probably implemented but not tested, as this will keep # character, which will of course then not work.
> 
> Please check for other cases of this, fix them and retest

I implemented, tested, but did not think it should not stay. I saw it but I never noticed they should not be inside ResourceLocation 😅

Reply:
The game crashes if you try to repair armor with # in resource location

Reply:
When testing, I compared what I had with what you asked me and what I was getting when adding a tag (as I already fully tested it). Then, I compared both codes to see if something was missing (as they use the exact MC code), but I never realized the `#` was not supposed to be generated. This is what I meant.